### PR TITLE
Fix publish error (no 'prefix' in str format)

### PR DESCRIPTION
**Pull request by alikins**
Fixes: ansible/galaxy-dev#163

### CODE REVIEW

File: galaxy_api/api/v3/viewsets.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
 from galaxy_api.common import pulp
 from galaxy_api.api import permissions, models
 
+import logging
```
Comment (romanblanco) [inline galaxy_api/api/v3/viewsets.py]: @alikins I've already created a PR adding `logging`, the order should be updated: https://github.com/ansible/galaxy-api/pull/114#discussion_r342482673

Feel free to close my PR if this is merged first.

**Comment by romanblanco** [comment]
> It's still broken
> 
> ```
> 
>   |  
> -- | --
> Response payload |  
> IntegrityError at /api/automation-hub/v3/artifacts/collections/ null value in column "task_id" violates not-null constraint DETAIL:  Failing row contains (null, null, collection_demo, 1.0.26, 1).   Request Method: POST Request URL: http://host.docker.internal:5001/api/automation-hub/v3/artifacts/collections/ Django Version: 2.2.5 Python Executable: /venv/bin/python3.6 Python Version: 3.6.8 Python Path: ['/venv/bin', '/usr/lib64/python36.zip', '/usr/lib64/python3.6', '/usr/lib64/python3.6/lib-dynload', '/venv/lib64/python3.6/site-packages', '/code/galaxy-api', '/venv/lib/python3.6/site-packages'] Server time: Wed, 6 Nov 2019 13:30:17 +0000 Installed Applications: ['django.contrib.admin',  'django.contrib.auth',  'django.contrib.contenttypes',  'django.contrib.messages',  'django.contrib.sessions',  'django.contrib.staticfiles',  'rest_framework',  'django_filters',  'galaxy_api.api',  'galaxy_api.auth'] Installed Middleware: ['django.middleware.security.SecurityMiddleware',  'django.middleware.common.CommonMiddleware',  'django.middleware.csrf.CsrfViewMiddleware',  'django.middleware.clickjacking.XFrameOptionsMiddleware',  'django.contrib.sessions.middleware.SessionMiddleware',  'django.contrib.auth.middleware.AuthenticationMiddleware',  'django.contrib.messages.middleware.MessageMiddleware']   Traceback:  File "/venv/lib64/python3.6/site-packages/django/db/backends/utils.py" in _execute   84.                 return self.cursor.execute(sql, params)  The above exception (null value in column "task_id" violates not-null constraint DETAIL:  Failing row contains (null, null, collection_demo, 1.0.26, 1). ) was the direct cause of the following exception:  File "/venv/lib64/python3.6/site-packages/django/core/handlers/exception.py" in inner   34.             response = get_response(request)  File "/venv/lib64/python3.6/site-packages/django/core/handlers/base.py" in _get_response   115.                 response = self.process_exception_by_middleware(e, request)  File "/venv/lib64/python3.6/site-packages/django/core/handlers/base.py" in _get_response   113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)  File "/venv/lib64/python3.6/site-packages/django/views/decorators/csrf.py" in wrapped_view   54.         return view_func(*args, **kwargs)  File "/venv/lib64/python3.6/site-packages/django/views/generic/base.py" in view   71.             return self.dispatch(request, *args, **kwargs)  File "/venv/lib64/python3.6/site-packages/rest_framework/views.py" in dispatch   505.             response = self.handle_exception(exc)  File "/venv/lib64/python3.6/site-packages/rest_framework/views.py" in handle_exception   465.             self.raise_uncaught_exception(exc)  File "/venv/lib64/python3.6/site-packages/rest_framework/views.py" in raise_uncaught_exception   476.         raise exc  File "/venv/lib64/python3.6/site-packages/rest_framework/views.py" in dispatch   502.             response = handler(request, *args, **kwargs)  File "/code/galaxy-api/galaxy_api/api/v3/viewsets.py" in post   200.             version=data['filename'].version,  File "/venv/lib64/python3.6/site-packages/django/db/models/manager.py" in manager_method   82.                 return getattr(self.get_queryset(), name)(*args, **kwargs)  File "/venv/lib64/python3.6/site-packages/django/db/models/query.py" in create   422.         obj.save(force_insert=True, using=self.db)  File "/venv/lib64/python3.6/site-packages/django/db/models/base.py" in save   741.                        force_update=force_update, update_fields=update_fields)  File "/venv/lib64/python3.6/site-packages/django/db/models/base.py" in save_base   779.                 force_update, using, update_fields,  File "/venv/lib64/python3.6/site-packages/django/db/models/base.py" in _save_table   870.             result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)  File "/venv/lib64/python3.6/site-packages/django/db/models/base.py" in _do_insert   908.                                using=using, raw=raw)  File "/venv/lib64/python3.6/site-packages/django/db/models/manager.py" in manager_method   82.                 return getattr(self.get_queryset(), name)(*args, **kwargs)  File "/venv/lib64/python3.6/site-packages/django/db/models/query.py" in _insert   1186.         return query.get_compiler(using=using).execute_sql(return_id)  File "/venv/lib64/python3.6/site-packages/django/db/models/sql/compiler.py" in execute_sql   1335.                 cursor.execute(sql, params)  File "/venv/lib64/python3.6/site-packages/django/db/backends/utils.py" in execute   99.             return super().execute(sql, params)  File "/venv/lib64/python3.6/site-packages/django/db/backends/utils.py" in execute   67.         return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)  File "/venv/lib64/python3.6/site-packages/django/db/backends/utils.py" in _execute_with_wrappers   76.         return executor(sql, params, many, context)  File "/venv/lib64/python3.6/site-packages/django/db/backends/utils.py" in _execute   84.                 return self.cursor.execute(sql, params)  File "/venv/lib64/python3.6/site-packages/django/db/utils.py" in __exit__   89.                 raise dj_exc_value.with_traceback(traceback) from exc_value  File "/venv/lib64/python3.6/site-packages/django/db/backends/utils.py" in _execute   84.                 return self.cursor.execute(sql, params)  Exception Type: IntegrityError at /api/automation-hub/v3/artifacts/collections/ Exception Value: null value in column "task_id" violates not-null constraint DETAIL:  Failing row contains (null, null, collection_demo, 1.0.26, 1).
> ```
> 
> Why aren't we using the same endpoint as before?

had the same issue as David (see nr.2 in "I've tried" section of https://github.com/ansible/galaxy-dev/issues/163 issue)

File: galaxy_api/api/v3/viewsets.py (hunk 1 above)
Comment (alikins) [reply galaxy_api/api/v3/viewsets.py]: @romanblanco I merged #114, and rebased this pr

**Comment by alikins** [comment]
> Why aren't we using the same endpoint as before?

@newswangerd As far as I know, we are using the same endpoint?  What was it before?

Pulp exposes pulp_ansible.app.viewsets.CollectionUploadViewSet at '/ansible/collections'
https://pulp-ansible.readthedocs.io/en/latest/restapi.html#operation/ansible_collections_create

**Comment by alikins** [comment]
@newswangerd @romanblanco I just updated and rebuilt everything and now I'm seeing the task_id null error as well.

Very verbose logs at http://pastebin.test.redhat.com/811942

**Comment by alikins** [comment]
Looks like GET to the task resource is setting task_id None, which causes the galaxy-api CollectionImport creation to fail because of the null task_id.

@newswangerd  Wasn't this previous broken and fixed on the pulp end? Feels like a regression.

```
pulp-api_1               | 2019-11-06 16:08:49,634| django.server basehttp:log_message:154 35 140167848826624 INFO| "GET /pulp/api/v3/tasks/6e500795-d56a-4507-a040-5985ca795e93/ HTTP/1.1" 200 489
galaxy-api_1             | DEBUG 2019-11-06 16:08:49,641 galaxy_api.api.v3.viewsets viewsets:post:192 33 Thread-6 task_detail: {'created_at': None,
galaxy-api_1             |  'error': None,
galaxy-api_1             |  'finished_at': None,
galaxy-api_1             |  'id': None,
galaxy-api_1             |  'messages': None,
galaxy-api_1             |  'started_at': datetime.datetime(2019, 11, 6, 16, 8, 49, 510373, tzinfo=tzlocal()),
galaxy-api_1             |  'state': 'running',
galaxy-api_1             |  'updated_at': None}
galaxy-api_1             | INFO 2019-11-06 16:08:49,642 galaxy_api.api.v3.viewsets viewsets:post:195 33 Thread-6 Publishing of artifact alikins-test_collection-0.0.112.tar.gz to namespace=alikins by user=alikins created pulp import task_id=None
postgres_1               | 2019-11-06 16:08:49.643 UTC [98] ERROR:  null value in column "task_id" violates not-null constraint
postgres_1               | 2019-11-06 16:08:49.643 UTC [98] DETAIL:  Failing row contains (null, null, test_collection, 0.0.112, 1).
postgres_1               | 2019-11-06 16:08:49.643 UTC [98] STATEMENT:  INSERT INTO "galaxy_api_collectionimport" ("task_id", "created_at", "namespace_id", "name", "version") VALUES (NULL, NULL, 1, 'test_collection', '0.0.112')

```

**Comment by alikins** [comment]
@newswangerd @romanblanco @daviddavis  

The similar previous issue:

https://github.com/ansible/galaxy-dev/issues/124
https://pulp.plan.io/issues/5569
https://github.com/pulp/pulp_ansible/pull/228

**Comment by alikins** [comment]
@newswangerd @romanblanco updated, should work now. 

The publish endpoint path introduced in f78082d009398d9415262a51c8b0d0cbd8fd7002 was wrong and broke things.

File: galaxy_api/api/v3/viewsets.py [hunk 2]
```diff
@@ -154,11 +154,11 @@ def post(self, request, *args, **kwargs):
             )
 
         self.check_object_permissions(request, namespace)
-
         api = pulp.get_client()
-        url = '{host}/{prefix}{path}'.format(
+        url = '{host}/{prefix}/{path}'.format(
             host=api.configuration.host,
-            path='ansible/collections/',
+            path='v3/artifacts/collections/',
```
Comment (cutwater) [inline galaxy_api/api/v3/viewsets.py]: If path component is static, it can be included in template string.

**Pull request merged**